src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/TransactionForRpc.cs

```diff
@@ -80,21 +80,20 @@ internal static void RegisterTransactionType<T>() where T : TransactionForRpc, I
 
         public override TransactionForRpc? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            using var document = JsonDocument.ParseValue(ref reader);
-
-            // TODO: For some reason we might get an object wrapped in a String
-            using JsonDocument jsonObject = document.RootElement.ValueKind == JsonValueKind.String
-                ? JsonDocument.Parse(document.RootElement.GetString()!)
-                : document;
-
-            TxType discriminator = DefaultTxType;
-            if (jsonObject.RootElement.TryGetProperty("type", out JsonElement typeProperty))
+            if (reader.ValueIsEscaped)
             {
-                discriminator = (TxType?)typeProperty.Deserialize(typeof(TxType), options) ?? DefaultTxType;
+                string escapedJson = JsonSerializer.Deserialize<string>(ref reader);
```

Comment:
Do we have to allocate the string?

Reply:
I think we do. I tried to do the following to avoid the allocation:

```c#
if (reader.ValueIsEscaped)
{
    reader.Skip();
    var jsonObject = new Utf8JsonReader(reader.ValueSpan);
    return JsonSerializer.Deserialize<Player>(ref jsonObject, options);
}
```

We "skip" the string but we try to deserialize the bytes that correspond to it. This fails in the object deserialization due to the quotes still being escaped as `\u0022` (see https://github.com/dotnet/runtime/issues/76130 for a similar issue).
It seems like the solution is either:
- Prevent the escaping at the origin
- Do a double parsing, first as a string, then as an object.

Reply:
Does the escaping thing is only because of how we double serialize params in the test? I think we can fix that.

Reply:
That seems to be the case. I've made some changes to our testing code to avoid this situation, but the solution is a bit hacky in my opinion.